sklearn/tree/tests/test_tree.py

```diff
@@ -2026,10 +2029,8 @@ def test_decision_tree_regressor_sample_weight_consistency(criterion):
     """Test that the impact of sample_weight is consistent."""
     tree_params = dict(criterion=criterion)
     tree = DecisionTreeRegressor(**tree_params, random_state=42)
-    for kind in ["zeros", "ones"]:
-        check_sample_weights_invariance(
-            "DecisionTreeRegressor_" + criterion, tree, kind="zeros"
-        )
+    check_unit_sample_weights("DecisionTreeRegressor_" + criterion, tree)
+    check_sample_weights_invariance("DecisionTreeRegressor_" + criterion, tree)
```

Comment:
@adrinjalali what do you think we should do here? Shall we register all criterions as separate instances as made possible in #29824? Or shall we keep this test as is?

Trying to leverage #29824 might significantly increase the runtime of common tests for little benefit for most common checks (e.g. input validation checks that are independent of the criterion).

Reply:
If you add them to `INIT_PARAMS`, then all tests are run for these estimators, but if you add them to `PER_ESTIMATOR_CHECK_PARAMS`, then only those tests are run multiple times and not the whole test suite, which would be equivalent to this current test. So both are possible, which one to choose is your judgement call.

Reply:
@ogrisel yes. I would even remove `test_decision_tree_regressor_sample_weight_consistency` as it is now covered by the new `check_sample_weights_invariance`